chrome/content/zotero/xpcom/connector/server_connector.js

```diff
@@ -1896,13 +1896,18 @@ Zotero.Server.Connector.Request.prototype = {
 			return [400, 'text/plain', 'Unsupported scheme'];
 		}
 
-		if (Zotero.Server.Connector.Request.validateHosts
-				&& !Zotero.Server.Connector.Request.allowedHosts.includes(uri.host)) {
-			return [
-				400,
-				'text/plain',
-				`Unsupported URL: host was ${uri.host}, expected one of [${Zotero.Server.Connector.Request.allowedHosts.join(', ')}]`
-			];
+		if (Zotero.Server.Connector.Request.enableValidation) {
+			if (!Zotero.Server.Connector.Request.allowedHosts.includes(uri.host)) {
+				return [
+					400,
+					'text/plain',
+					`Unsupported URL: host was ${uri.host}, expected one of [${Zotero.Server.Connector.Request.allowedHosts.join(', ')}]`
+				];
+			}
+
+			if (req.headers['User-Agent'] && !req.headers['User-Agent'].startsWith('Mozilla/')) {
```

Comment:
Should be `!req.headers['User-Agent'] || !req.headers['User-Agent'].startsWith('Mozilla/')`